cg/meta/transfer/external_data.py

```diff
@@ -0,0 +1,202 @@
+        )
+        all_fastq_in_folder: List[Path] = self.get_all_fastq(sample_folder=fastq_folder)
+        return all_fastq_in_folder
+
+    def create_hk_bundle(
+        self, bundle_data: dict, bundle_name: str, dry_run: bool
+    ) -> Optional[Tuple[Bundle, Version]]:
+        """Creates and returns a housekeeper bundle and the new version"""
+        if dry_run:
+            LOG.info(
+                "Would have created bundle %s in housekeeper, but this is dry-run", bundle_name
+            )
+            return
+        bundle_result: Tuple[Bundle, Version] = self.housekeeper_api.add_bundle(
+            bundle_data=bundle_data
+        )
+        return bundle_result
+
+    def check_fastq_md5sums(self, fastq_paths: List[Path]) -> List[Path]:
+        failed_sums: List[Path] = []
+        for path in fastq_paths:
+            if Path(str(path) + ".md5").exists():
+                given_md5sum: str = extract_md5sum(md5sum_file=Path(str(path) + ".md5"))
+                if not check_md5sum(file_path=path, md5sum=given_md5sum):
+                    failed_sums.append(path)
+        return failed_sums
+
+    def add_files_to_bundles(
+        self, fastq_paths: List[Path], last_version: Version, lims_sample_id: str
+    ):
+        """Adds the given fastq files to the the hk-bundle"""
+        for path in fastq_paths:
+            LOG.info("Adding path %s to bundle %s in housekeeper" % (path, lims_sample_id))
+            self.housekeeper_api.add_file(path=path, version_obj=last_version, tags=HK_FASTQ_TAGS)
+
+    def add_transfer_to_housekeeper(self, ticket_id: int, dry_run: bool = False) -> None:
+        """Creates sample bundles in housekeeper and adds the files corresponding to the ticket to the bundle"""
+        cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
+        cust: str = self.status_db.get_customer_id_from_ticket(ticket_id=ticket_id)
+        for case in cases:
+            links = case.links
+            for link in links:
+                lims_sample_id: str = link.sample.internal_id
+                paths: List[Path] = self.get_all_paths(lims_sample_id=lims_sample_id, customer=cust)
+                if dry_run:
+                    LOG.info(
+                        "Would have added %s to housekeeper and linked associated files, but this is dry-run",
+                        lims_sample_id,
+                    )
+                    return
+                last_version: Version = self.housekeeper_api.get_create_version(
+                    bundle=lims_sample_id
+                )
+                fastq_paths_to_add: List[Path] = self.housekeeper_api.check_bundle_files(
+                    file_paths=paths,
+                    bundle_name=lims_sample_id,
+                    last_version=last_version,
+                    tags=HK_FASTQ_TAGS,
+                )
+                failed_sums: List[Path] = self.check_fastq_md5sums(fastq_paths_to_add)
```

Comment:
do we want to check sums even in dry run or not? because in dry run it will never be checked